### PR TITLE
Minify JS in wasm mode too

### PR DESCRIPTION
**Pull request by kripken**
In small projects, the JS side is significant so worth minifying when possible. That we didn't do this already was an oversight.

### CODE REVIEW

File: emcc.py [hunk 1]
```diff
@@ -1273,7 +1273,6 @@ def check(input_file):
         shared.Settings.ASMJS_CODE_FILE = os.path.basename(asm_target)
 
         shared.Settings.ASM_JS = 2 # when targeting wasm, we use a wasm Memory, but that is not compatible with asm.js opts
-        debug_level = max(1, debug_level) # keep whitespace readable, for asm.js parser simplicity
```
Comment (kripken) [inline emcc.py]: this was an old hack, no longer needed

File: emcc.py [hunk 2]
```diff
@@ -2200,6 +2199,21 @@ def do_minify(): # minifies the code. this is also when we do certain optimizati
               os.unlink(js_target) # we don't need the js, it can just confuse
               os.unlink(asm_target) # we don't need the asm.js, it can just confuse
             sys.exit(0) # and we are done.
+        if opt_level >= 2:
```
Comment (dschuff) [inline emcc.py]: Does this optimize stage do anything different than for the asm.js case? Is there any way the code can be shared?

Comment (kripken) [inline emcc.py]: It's basically the same. The difference is that for asm.js, we do it in js_optimizer.py, which first splits out the asm.js, then can run passes on that code separately from the shell JS. It also coordinates those opts, as some do affect the shell, so there is a bit of complexity there. I didn't see a natural way to share this code. Instead, my hope is that at some point we'll deprecate asm.js support (leaving it to older versions), and get rid of the js_optimizer code path.

File: emcc.py [hunk 3]
```diff
@@ -2200,6 +2199,21 @@ def do_minify(): # minifies the code. this is also when we do certain optimizati
...
+          if JSOptimizer.cleanup_shell or JSOptimizer.minify_whitespace or use_closure_compiler:
+            misc_temp_files.note(final)
+            shutil.move(js_target, final)
+            if DEBUG: save_intermediate('preclean', 'js')
+            if use_closure_compiler:
+              if DEBUG: print >> sys.stderr, 'running closure on shell code'
+              final = shared.Building.closure_compiler(final, pretty=not JSOptimizer.minify_whitespace)
+            else:
+              assert JSOptimizer.cleanup_shell
+              if DEBUG: print >> sys.stderr, 'running cleanup on shell code'
```
Comment (dschuff) [inline emcc.py]: should this be removed or use the regular emscripten logging infrastructure?

**Pull request merged**